src/mesh/data/gridfromfile.cxx

```diff
@@ -98,7 +104,13 @@ bool GridFile::get(Mesh *UNUSED(m), BoutReal &rval, const string &name) {
   if(!file->is_valid())
     throw BoutException("File cannot be read");
   
-  return file->read(&rval, name);
+  bool success = file->read(&rval, name);
+    if(success){
```

Comment:
I do not understand what is wrong here? 

Furthermore, I would suggest that if statements without braces are not allowed. They can easily cause bugs and misreading the code?